src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    element.addEventListener(eventName, (e) => {
+      e.preventDefault();
```

Comment:
Using `e.preventDefault()` is typically used to prevent the default action of an event. Ensure that this is necessary for the events being listened to, especially for non-form elements like `loginField` and `passwordField`.